src/rest-server/src/util/manager/user/crudK8sSecret.js

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const User = require('./user');
+const axios = require('axios');
+const {readFileSync} = require('fs');
+const {Agent} = require('https');
+
+function initConfig(apiServerUri, namespace, option) {
+  const config = {
+    'apiServerUri': apiServerUri,
+    'namespace': namespace,
+    'reqeustConfig': {
+      'baseURL': `${apiServerUri}/api/v1/namespaces/`,
+      'maxRedirects': 0,
+    },
+  };
+  if ('k8sAPIServerCaFile' in option) {
+    const ca = readFileSync(option.k8sAPIServerCaFile);
+    config.reqeustConfig.httpsAgent = new Agent({ca});
+  }
+  if ('k8sAPIServerTokenFile' in option) {
+    const token = readFileSync(option.k8sAPIServerTokenFile, 'ascii');
+    config.reqeustConfig.headers = {Authorization: `Bearer ${token}`};
+  }
+}
+
+function getSecretRootUri(config) {
+  return `${config.namespace}/secrets`;
+}
+
+async function read(key, config) {
```

Comment:
need some documentation comments about the config param

Reply:
Done